modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/WebColorFieldSkin.java

```diff
@@ -99,23 +99,18 @@ protected void updateText() {
 
     @Override
     protected void updateValue() {
-        if (noChangeInValue) return;
         Color value = ((WebColorField) control).getValue();
-        String text = getTextField().getText() == null ? "" : getTextField().getText().trim().toUpperCase(Locale.ROOT);
-        if (text.matches("#[A-F0-9]{6}") || text.matches("[A-F0-9]{6}")) {
-            try {
-                Color newValue = (text.charAt(0) == '#')? Color.web(text) : Color.web("#"+text);
-                if (!newValue.equals(value)) {
-                    ((WebColorField) control).setValue(newValue);
-                } else {
-                    // calling setText results in updateValue - so we set this flag to true
-                    // so that when this is true updateValue simply returns.
-                    noChangeInValue = true;
-                    getTextField().setText(Utils.formatHexString(newValue));
-                    noChangeInValue = false;
+        String text = getTextField().getText() == null ? "" : getTextField().getText().trim();
+        if (PATTERN.matcher(text).matches()) {
+            Color newValue = (text.charAt(0) == '#') ? Color.web(text) : Color.web("#" + text);
+            if (!newValue.equals(value)) {
+                ((WebColorField) control).setValue(newValue);
+            } else {
+                String newText = Utils.formatHexString(newValue);
+
+                if (!newText.equals(text)) {
+                    getTextField().setText(newText);
```

Comment:
I would recommend to keep only listener specific change in this PR. If the pattern matching can be skipped please revert those. I would be glad to review in a separate PR.

Reply:
Removed